include/flatbuffers/flatbuffers.h

```diff
@@ -712,15 +901,23 @@ FLATBUFFERS_FINAL_CLASS
  public:
   /// @brief Default constructor for FlatBufferBuilder.
   /// @param[in] initial_size The initial size of the buffer, in bytes. Defaults
-  /// to`1024`.
-  /// @param[in] allocator A pointer to the `simple_allocator` that should be
-  /// used. Defaults to `nullptr`, which means the `default_allocator` will be
-  /// be used.
+  /// to `1024`.
+  /// @param[in] allocator An `Allocator` to use. Defaults to a new instance of
+  /// a `DefaultAllocator`.
+  /// @param[in] steal_allocator Whether to steal the allocator. Defaults to
+  /// `false`, unless it's a newly allocated `DefaultAllocator`.
   explicit FlatBufferBuilder(uoffset_t initial_size = 1024,
-                             const simple_allocator *allocator = nullptr)
-      : buf_(initial_size, allocator ? *allocator : default_allocator),
-        nested(false), finished(false), minalign_(1), force_defaults_(false),
-        dedup_vtables_(true), string_pool(nullptr) {
+                             Allocator *allocator = nullptr,
+                             bool steal_allocator = false)
+    : buf_(initial_size,
+           allocator ? allocator : new DefaultAllocator(),
```

Comment:
This was introduced when we decided we didn't want templates and went to inheritance, which means an extra level of indirection, and we can't store the allocator as a member. I can modify the default case to point to a static `DefaultAllocator` since it's stateless.